System.Containers.Tasks/build/System.Containers.Tasks.targets

```diff
@@ -3,43 +3,68 @@
         <TargetFrameworkVersion>6.0</TargetFrameworkVersion>
     </PropertyGroup>
 
-    <!-- Required targets dependencies -->
-    <PropertyGroup>
-        <PublishContainerDependsOn>
-            Build;
-            ComputeContainerConfig
-        </PublishContainerDependsOn>
-    </PropertyGroup>
-
-    <Target Name="ComputeContainerConfig" AfterTargets="Build;_CopyResolvedFilesToPublishPreserveNewest">
-        <!-- Find everything in the publish directory -->
-        <ItemGroup>
-            <_BuildOutputsNonRelative Include="$(MSBuildProjectDirectory)\$(PublishDir)**\*.*"/>
-            <BuildOutputs Include="%(_BuildOutputsNonRelative.FullPath)"/>
-        </ItemGroup>
+    <Target Name="ComputeContainerConfig">
+        <!-- Compute private defaults -->
+        <PropertyGroup>
+            <_ContainerBaseRegistry Condition="'$(_ContainerBaseRegistry)' == ''">https://mcr.microsoft.com</_ContainerBaseRegistry>
+            <_ContainerBaseImageName Condition="'$(_ContainerBaseImageName)' == '' and @(ProjectCapability->AnyHaveMetadataValue('Identity', 'AspNetCore'))">dotnet/aspnet</_ContainerBaseImageName>
+            <_ContainerBaseImageName Condition="'$(_ContainerBaseImageName)' == ''">dotnet/runtime</_ContainerBaseImageName>
+            <_ContainerBaseImageTag Condition="'$(_ContainerBaseImageTag)' == ''">$(_TargetFrameworkVersionWithoutV)</_ContainerBaseImageTag>
+        </PropertyGroup>
 
         <!-- Container Defaults -->
         <PropertyGroup>
-            <ContainerBaseImageName Condition="'$(ContainerBaseImageName)' == ''"></ContainerBaseImageName>
-            <ContainerBaseImageTag Condition="'$(ContainerBaseImageTag)' == '' and $([MSBuild]::VersionGreaterThanOrEquals('$(TargetFrameworkVersion)', 6.0))">6.0</ContainerBaseImageTag>
-            <ContainerBaseImage Condition="'$(ContainerBaseImage)' == ''">$(ContainerBaseImageName):$(ContainerBaseImageTag)</ContainerBaseImage>
-            <ContainerInputRegistryURL Condition="'$(ContainerInputRegistryURL)' == ''">http://localhost:5010</ContainerInputRegistryURL>
-            <ContainerOutputRegistryURL Condition="'$(ContainerOutputRegistryURL)' == ''">http://localhost:5010</ContainerOutputRegistryURL>
-            <ContainerImageName Condition="'$(ContainerImageName)' == ''">$(AssemblyName.ToLower())</ContainerImageName>
+            <ContainerBaseImage Condition="'$(ContainerBaseImage)' == ''">$(_ContainerBaseRegistry)/$(_ContainerBaseImageName):$(_ContainerBaseImageTag)</ContainerBaseImage>
```

Comment:
Do these properties need to be assigned in the target, or could they be set at eval time to be a bit more obvious? I prefer the latter if this file is late enough in the import order (in the final SDK location as well as now via NuGet) to catch set-in-project properties.

Reply:
Some of them did, I think it was either `ContainerImageTag` or `ContainerImageName` because `AssemblyName` or `Version` wasn't set by the time this targets file was read.

I'm not sure what we get by making this an eval-time thing. Is this for diagnosing build issues?

Brain-dumping some thoughts on that: If these were set during eval-time, we'd see them in the overall evaluation of a project. That's definitely an advantage over needing to dig into the project and this target to see what the values were.

We could theoretically break out only the properties that don't have timing issues into eval-time, my main concern there would be timing (release timing, not build timing)

Reply:
The `Version` property is set fairly late - the best place I've found for it in my explorations (because my CHANGELOG-versioning targets needed it) was doing a `"BeforeBuild"` target dependency. So evaluation won't work for anything that depends on `Version`, at least.

Reply:
Also I've spoken with the VS team and they're perfectly fine with calling a specific target to get any container-specific metadata that's required, they're not hung up on eval-only data.

Reply:
I think we'd have to call the target anyway so I guess there's no huge reason to prefer eval-time.